src/api/posts/test/posts.test.js

```diff
@@ -0,0 +1,166 @@
+
+  test('default number of items should be returned', async (done) => {
+    const res = await request(app).get('/');
+
+    expect(res.status).toBe(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    expect(res.body.length).toBeGreaterThan(0);
+    expect(res.body instanceof Array).toBe(true);
+    done();
+  });
+
+  test('requests a specific number of items', async (done) => {
+    const res = await request(app).get('/?per_page=50');
+
+    expect(res.status).toEqual(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    expect(res.body.length).toBe(requestedItems);
+    expect(res.body instanceof Array).toBe(true);
+    done();
+  });
+
+  test('requests more items than the limit set by the server', async (done) => {
+    const res = await request(app).get('/?per_page=150');
+
+    expect(res.status).toEqual(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    expect(res.body.length).toBe(maxItems);
+    expect(res.body instanceof Array).toBe(true);
+    done();
+  });
+
+  test('request posts with both query params', async (done) => {
+    const res = await request(app).get('/?page=2&per_page=50');
+    expect(res.status).toEqual(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    expect(res.body.length).toBe(requestedItems);
+    expect(res.body instanceof Array).toBe(true);
+    done();
+  });
+
+  test('request posts with both only page param', async (done) => {
+    const res = await request(app).get('/?page=3');
+    expect(res.status).toEqual(200);
+    expect(res.get('Content-type')).toContain('application/json');
+    expect(res.get('X-Total-Count')).toBe(createdItems.toString());
+    // This will depend on the env value, so as long as we get back something.
+    expect(res.body.length).toBeGreaterThan(0);
+    expect(res.body instanceof Array).toBe(true);
+    done();
+  });
+
+  test('request posts with non-integer page param', async (done) => {
+    const res = await request(app).get(`/?page=${nonInteger}`);
+    expect(res.status).toEqual(400);
+    done();
+  });
+  test('request posts with non-integer per_page param', async (done) => {
+    const res = await request(app).get(`/?per_page=${nonInteger}`);
+    expect(res.status).toEqual(400);
+    done();
+  });
+});
+
+describe('test /posts/:id responses', () => {
+  // An array of keys used to test whether an id that doesn't exist cannot be found
+  const missingGuid = 'http://missing-guid';
+
+  test('A post with an id should be returned', async (done) => {
+    const res = await request(app).get('/');
+    const postIDRes = await request(app).get(`/${res.body[0].id}`);
```

Comment:
Write another test like this, but use the `url` you get vs. the `id`, to make sure both work.